state/tx/indexer/kv.go

```diff
@@ -0,0 +1,55 @@
+package indexer
+
+import (
+	"bytes"
+	"fmt"
+
+	db "github.com/tendermint/go-db"
+	"github.com/tendermint/go-wire"
+	"github.com/tendermint/tendermint/types"
+)
+
+// KV is a simplest possible indexer, backed by Key-Value storage (levelDB).
+// It could only index transaction by its identifier.
+type KV struct {
```

Comment:
So really the correct thing would be a package for each implementation with a struct called `Indexer` ?